### ISSUE TITLE
Go lang api connectivity for kubedog

### ISSUE THREAD

**Issue by JeffreySunjay4044** [JeffreySunjay4044/kubedog#4]
To do :
Go lang based micro service architecture.
Write task executor which can run tasks run in parallel. The underlying task can be a data pull or data push operation.
Go lang slack connectivity with authentication plugin

Notes:
Please pick up a new PR for each change
Lets keep it to the opensource community standards in naming variables and functions.
Place this code under api folder in kubedog
